main.cpp

```diff
@@ -56,46 +56,73 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(int v, const char* n) : value(v), name(n) {} //1
+    int value; //2
+    std::string name; //3
 };
 
-struct <#structName1#>                                //4
+struct Comparison                               //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if( a != nullptr && b != nullptr )
+        {
+            if( a->value < b->value ) return a;
+            if( a->value > b->value ) return b;
+        }
         return nullptr;
     }
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float a { 0 }, b { 0 };
+    float updateValues(float* updated)      //12
     {
-        
+        if( updated != nullptr )
+        {
+            std::cout << "U's a value: " << this->a << std::endl;
+            this->a = *updated;
+            std::cout << "U's a updated value: " << this->a << std::endl;
+            while( std::abs(this->b - this->a) > 0.001f )
+            {
+                this->b += 1.5f;
+            }
+            std::cout << "U's b updated value: " << this->b << std::endl;
+            return this->b * this->a;
+        }
+        else
+        {
+            std::cout << "'updated' is nullptr" <<  std::endl;
+            return 0;
+        }
     }
 };
 
-struct <#structname2#>
+struct P
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float updateValues(U* that, float* updated)        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        if( that != nullptr && updated != nullptr )
+        {    
+            std::cout << "U's a value: " << that->a << std::endl;
+            that->a = *updated;
+            std::cout << "U's a updated value: " << that->a << std::endl;
+            while( std::abs(that->b - that->a) > 0.001f )
+            {
+                /*
+                write something that makes the distance between that-><#name2#> and that-><#name1#> get smaller
+                */
+                that->b += 1.5f;
+            }
```

Comment:
adding `1.5` to `b` does not make the distance between `a` and `b` get smaller
- [ ] revise